### PR TITLE
Make HTTP Service Inside the WebSubHub Listener an Isolated Object

### PR DESCRIPTION
**Pull request by ayeshLK**
Fixes [#1390](https://github.com/ballerina-platform/ballerina-standard-library/issues/1390)

## Checklist
- [X] Linked to an issue
- [X] Updated the changelog
~~Added tests~~ N/A

### CODE REVIEW

File: websubhub-ballerina/http_service.bal [hunk 1]
```diff
@@ -18,56 +18,40 @@ import ballerina/http;
...
-                    self.isDeregisterAvailable = true;
-                }
-            }
-        }
+        self.isSubscriptionAvailable = isMethodAvailable("onSubscription", methodNames);
+        self.isSubscriptionValidationAvailable = isMethodAvailable("onSubscriptionValidation", methodNames);
+        self.isUnsubscriptionAvailable = isMethodAvailable("onUnsubscription", methodNames);
+        self.isUnsubscriptionValidationAvailable = isMethodAvailable("onUnsubscriptionValidation", methodNames);
+        self.isRegisterAvailable = isMethodAvailable("onRegisterTopic", methodNames);
+        self.isDeregisterAvailable = isMethodAvailable("onDeregisterTopic", methodNames);
```
Comment (shafreenAnfar) [inline websubhub-ballerina/http_service.bal:54]: Any reason for this change ?

File: websubhub-native/src/main/java/io/ballerina/stdlib/websubhub/RequestHandler.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package io.ballerina.stdlib.websubhub;
+
+import io.ballerina.runtime.api.values.BObject;
+
+/**
+ * {@code RequestHandler} is a wrapper object used for service method execution.
+ */
+public class RequestHandler {
```
Comment (shafreenAnfar) [inline websubhub-native/src/main/java/io/ballerina/stdlib/websubhub/RequestHandler.java:8]: Any reason for this ?

Comment (ayeshLK) [reply websubhub-native/src/main/java/io/ballerina/stdlib/websubhub/RequestHandler.java:8]: * We could not initialize an object which is not isolated in an isolated-object. In this case `websubhub:Service` was not isolated so we need to have a wrapper around it so we could just invoke remote methods in the `websubhub:Service` class using that.

Comment (ayeshLK) [reply websubhub-native/src/main/java/io/ballerina/stdlib/websubhub/RequestHandler.java:25]: * Moving all the native method calls to `RequestHandler` as per the discussion.

File: websubhub-ballerina/hub_listener.bal [hunk 3]
```diff
@@ -60,15 +60,18 @@ public class Listener {
 
         string hubUrl = self.retrieveHubUrl(name);
         ServiceConfiguration? configuration = retrieveServiceAnnotations('service);
+        RequestHandler handler = new ();
```
Comment (shafreenAnfar) [inline websubhub-ballerina/hub_listener.bal:63]: Lets change this to RequestHandler handler = new ('service)

File: websubhub-ballerina/utils.bal [hunk 4]
```diff
@@ -20,22 +20,26 @@ import ballerina/http;
 import ballerina/uuid;
 import ballerina/regex;
 
+# Wrapper class used to execute sevice methods.
+isolated class RequestHandler {
+}
+
```
Comment (shafreenAnfar) [inline websubhub-ballerina/utils.bal:26]: I think this is better.

isolated class RequestHandler {
     isolated function callOnUnsubscriptionIntentVerifiedMethod(VerifiedUnsubscription msg, http:Headers headers) {} extern.
}

File: websubhub-ballerina/http_service.bal (hunk 1 above)
Comment (ayeshLK) [reply websubhub-ballerina/http_service.bal:54]: * Resolving since we have concluded to use this in our discussion.

**Review by shafreenAnfar — APPROVED** [review]

**Pull request merged**